tests/end_to_end/utils/docker_helper.py

```diff
@@ -0,0 +1,149 @@
+
+
+def check_docker_image():
+    """
+    Check if the docker image exists.
+    """
+    client = get_docker_client()
+    images = client.images.list(name=constants.DEFAULT_OPENFL_IMAGE)
+    if not images:
+        log.error(f"Image {constants.DEFAULT_OPENFL_IMAGE} does not exist")
+        raise Exception(f"Image {constants.DEFAULT_OPENFL_IMAGE} does not exist")
+    log.debug(f"Image {constants.DEFAULT_OPENFL_IMAGE} exists")
+
+
+def start_docker_container(
+    container_name,
+    workspace_path,
+    local_bind_path,
+    image=constants.DEFAULT_OPENFL_IMAGE,
+):
+    """
+    Start the docker container with provided name.
+    Args:
+        container_name: Name of the container
+        workspace_path: Workspace path
+        local_bind_path: Local bind path
+        image: Docker image to use
+    Returns:
+        container: Docker container object
+    """
+    try:
+        client = get_docker_client()
+
+        # Local bind path
+        local_participant_path = os.path.join(local_bind_path, container_name, "workspace")
+
+        # Docker container bind path
+        docker_participant_path = f"{workspace_path}/{container_name}/workspace"
+
+        volumes = {
+            local_participant_path: {"bind": docker_participant_path, "mode": "rw"},
+        }
+
+        # Start a container from the image
+        container = client.containers.run(
+            image,
+            detach=True,
+            user="root",
+            auto_remove=False,
+            tty=True,
+            name=container_name,
+            network="openfl",
+            volumes=volumes,
+            environment={
+                "WORKSPACE_PATH": docker_participant_path,
+                "NO_PROXY": "aggregator",
+                "no_proxy": "aggregator",
+            },
+            use_config_proxy=False,  # Do not use proxy for docker container
+        )
+        log.info(f"Container for {container_name} started with ID: {container.id}")
+
+    except Exception as e:
+        raise ex.DockerException(f"Error starting docker container: {e}")
+
+    return container
+
+
+@lru_cache(maxsize=50)
+def get_docker_client():
+    """
+    Get the Docker client.
+    Returns:
+        Docker client
+    """
+    try:
+        client = docker.DockerClient(base_url="unix://var/run/docker.sock")
+    except Exception as e:
+        raise ex.DockerException(f"Error getting docker client: {e}")
+    return client
+
+
+def cleanup_docker_containers():
+    """
+    Cleanup the docker containers meant for openfl.
+    """
+    log.debug("Cleaning up docker containers")
+
+    client = get_docker_client()
+
+    # List all containers related to openfl
+    agg_containers = client.containers.list(all=True, filters={"name": "aggregator"})
+    col_containers = client.containers.list(all=True, filters={"name": "collaborator*"})
+
+    # itp_tool_container = docker_client.containers.get(itp_tool_container_id)
```

Comment:
Is this comment relevant?

Reply:
No. I have removed it. Must be a wrong copy paste.